### PR TITLE
C# and Java client negotiateVersion cleanup

### PR DESCRIPTION
**Pull request by BrennanConroy**
* negotiateVersion is not included after negotiation occurs
* negotiateVersion is not added if it is already present
* connectionToken is not stored anywhere

### CODE REVIEW

**Review by halter73 — APPROVED** [review]

File: src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs [hunk 1]
```diff
@@ -343,7 +342,7 @@ private async Task SelectAndStartTransport(TransferFormat transferFormat, Cancel
                 }
 
                 // This should only need to happen once
-                var connectUrl = CreateConnectUrl(uri, _connectionToken);
+                var connectUrl = CreateConnectUrl(uri, negotiationResponse.ConnectionToken);
```
Comment (anurse) [inline src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs]: This URL isn't exposed anywhere, right?

Comment (BrennanConroy) [reply src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs]: It does get logged at the debug level

Comment (anurse) [reply src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs]: That's OK

Comment (anurse) [reply src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs]: Well, I think it is. We could also redact it from the logs.

@blowdart @GrabYourPitchforks do you have any thoughts here on if the connection token value should be suppressed from `Debug` level logs on the **Client**.

**Comment by blowdart** [comment]
Yes they should

________________________________

**Comment by BrennanConroy** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=366242&view=logs

**Pull request merged**